src/server/logging/log_format.js

```diff
@@ -111,30 +111,33 @@ export default class TransformObjStream extends Stream.Transform {
         'load'
       ]));
       data.message  = ansicolors.brightBlack('memory: ');
-      data.message += numeral(data.proc.mem.heapUsed).format('0.0b');
+      data.message += numeral(get(data, 'proc.mem.heapUsed')).format('0.0b');
       data.message += ' ';
       data.message += ansicolors.brightBlack('uptime: ');
-      data.message += numeral(data.proc.uptime).format('00:00:00');
+      data.message += numeral(get(data, 'proc.uptime')).format('00:00:00');
       data.message += ' ';
       data.message += ansicolors.brightBlack('load: [');
-      data.message += data.os.load.map(function (val) {
+      data.message += get(data, 'os.load', []).map(function (val) {
         return numeral(val).format('0.00');
       }).join(' ');
       data.message += ansicolors.brightBlack(']');
       data.message += ' ';
       data.message += ansicolors.brightBlack('delay: ');
-      data.message += numeral(data.proc.delay).format('0.000');
+      data.message += numeral(get(data, 'proc.delay')).format('0.000');
     }
     else if (data.type === 'error') {
       data.level = 'error';
-      data.message = event.error.message;
       data.error = serializeError(event.error);
       data.url = event.url;
+      const message =  get(event, 'error.message');
+      data.message = message || `Event Error stack: ${get(event, 'error.stack')}`;
     }
     else if (event.data instanceof Error) {
+      data.type = 'error';
       data.level = _.contains(event.tags, 'fatal') ? 'fatal' : 'error';
-      data.message = event.data.message;
       data.error = serializeError(event.data);
+      const message =  get(event, 'data.message');
+      data.message = message || `Event Data Error stack: ${get(event, 'data.stack')}`;
```

Comment:
**nit:** I think the `Event Data Error stack:` prefix here is unnecessary. Also, what would possibly cause there to be a stack when there is no message? It's generally done the other way around because stack traces are not produced in all situations, so `error.stack || error.message` would produce something everywhere even if stack traces are not supported.

Reply:
> `Event Data Error stack:` prefix here is unnecessary

I added that as a safety in case the stack is also undefined, that way we don't just log `undefined`, but have a label that we can use to work backwards and try to figure out what happened.

> It's generally done the other way around because stack traces are not produced in all situations, so error.stack || error.message would produce something everywhere even if stack traces are not supported.

If there is both a stack and a message, we wouldn't want to have just the stack right? I think this is something we need to figure out. What about concatenating the 2 fields instead of `or`-ing them?

Reply:
Well, stack starts with the message, so concatenating them is just messy. My point is more that if message isn't defined I don't think there's any reason to think that stack will be, so perhaps it would be better if we did something like:

```
data.message = message || `Unknown error (no message)`
```